app/src/main/java/ch/epfl/favo/gps/GpsTracker.java

```diff
@@ -11,32 +11,41 @@
 import android.os.IBinder;
 
 import androidx.core.content.ContextCompat;
-import androidx.fragment.app.FragmentActivity;
 
 import ch.epfl.favo.exception.NoPermissionGrantedException;
 import ch.epfl.favo.exception.NoPositionFoundException;
 import ch.epfl.favo.util.DependencyFactory;
 
-public class GpsTracker extends FragmentActivity implements LocationListener, IGpsTracker {
+public class GpsTracker implements LocationListener, IGpsTracker {
+  private static final int millisecToSce = 100;
```

Comment:
I find these variable names slightly confusing, maybe you can add comments where they're used to indicate how many seconds it takes to do a refresh/new location fetch etc

Reply:
I made two mistakes here, it should be` millisecToSec = 1000`, meaning convert unit from millisecond to second.
Thanks. And I add comments for other variables too.